src/core/server/plugins/plugins_system.test.ts

```diff
@@ -517,4 +536,71 @@ describe('start', () => {
     const log = logger.get.mock.results[0].value as jest.Mocked<Logger>;
     expect(log.info).toHaveBeenCalledWith(`Starting [2] plugins: [order-1,order-0]`);
   });
+
+  it('validates opensearch plugin installation', async () => {
+    [
+      createPlugin('order-1', { requiredOSPlugin: ['test-plugin'] }),
+      createPlugin('order-2'),
+    ].forEach((plugin, index) => {
+      jest.spyOn(plugin, 'setup').mockResolvedValue(`setup-as-${index}`);
+      jest.spyOn(plugin, 'start').mockResolvedValue(`started-as-${index}`);
+      pluginsSystem.addPlugin(plugin);
+    });
+
+    const opensearch = startDeps.opensearch;
+    opensearch.client.asInternalUser.cat.plugins.mockResolvedValueOnce({
+      body: [
+        {
+          name: 'node-1',
+          component: 'test-plugin',
+          version: 'v1',
+        },
+      ],
+    } as any);
```

Comment:
seems like this is duplicated a couple times - it's nice and explicit this way, but would it be cleaner to just use `mockResolvedValue` instead of `mockResolvedValueOnce` multiple times?